[ryanlimdx] iP

ryanlimdx opened the pull request:
# Woody
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Woody frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~Fast~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [X] Managing tasks
- [X] Find tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        new Duke("./data/data.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ryanlimdx committed:
renamed file and class with chatbot's new name

ryanlimdx committed:
echo

ryanlimdx committed:
added personality

ryanlimdx committed:
named file back to Duke

ryanlimdx committed:
added list to store user input

ryanlimdx committed:
added task class

ryanlimdx committed:
added mark/ unmark feature

ryanlimdx committed:
added todos, events and deadlines file. abstracted away Task string.

ryanlimdx committed:
added different commands for adding varying types of tasks

ryanlimdx committed:
conducted text ui testing

ryanlimdx committed:
added exception handling for adding and marking tasks. reorganised section on adding tasks

ryanlimdx committed:
added feature to delete tasks

ryanlimdx committed:
added ability to save previous commands

ryanlimdx committed:
Merge branch 'branch-Level-7'

ryanlimdx committed:
added dates and times

ryanlimdx committed:
Merge branch 'branch-Level-8'

ryanlimdx committed:
reorganise project structure for OOP

ryanlimdx committed:
reorganise project structure for OOP

ryanlimdx committed:
clean up

ryanlimdx committed:
Merge branch 'branch-A-MoreOOP'

ryanlimdx committed:
refactor into packages

ryanlimdx committed:
fix re-appending previous tasks

ryanlimdx committed:
refactor task write out of tasks

ryanlimdx committed:
clean up

ryanlimdx committed:
Merge branch 'A-Packages'

ryanlimdx committed:
Merge branch 'add-gradle-support'

ryanlimdx committed:
add JUnit tests

ryanlimdx committed:
Merge branch 'branch-A-JUnit'

ryanlimdx committed:
add JavaDocs for methods in the duke package (excluding utils)

ryanlimdx committed:
add JavaDocs for methods in the duke package (excluding utils)

ryanlimdx committed:
Merge branch 'branch-A-JavaDoc'

ryanlimdx committed:
keep content at a length limit of 90

ryanlimdx committed:
Merge branch 'branch-A-CodingStandard'

ryanlimdx committed:
Coding standards for egyptian braces and naming convention for constants

ryanlimdx committed:
add ability to search for tasks

ryanlimdx committed:
Merge branch 'branch-Level-9'

ryanlimdx committed:
add javadoc for searchTasks method

ryanlimdx committed:
add exception for find query having no description

ryanlimdx committed:
Add more accepted date formats

Users may be more familiar with just typing in the date.

The string is split to determine if time is specified. If it is,
datetime is used. Otherwise, it will be of date type.

Let's allow multiple input format.

ryanlimdx committed:
Merge branch 'branch-Level-8'

ryanlimdx committed:
Fix bug where an exception is thrown when processing date with no time input

ryanlimdx committed:
Merge branch 'branch-level-8'

ryanlimdx committed:
Add some content to README.md

ryanlimdx committed:
Fix styling of code

ryanlimdx committed:
Merge branch 'branch-A-CheckStyle'

ryanlimdx committed:
Set up JavaFx

ryanlimdx committed:
Add Command class

ryanlimdx committed:
Abstract commands into separate classes

ryanlimdx committed:
Clean up Parser class

ryanlimdx committed:
Clean up run method

ryanlimdx committed:
Clean up run() method

ryanlimdx committed:
Remove unnecessary variable in Command class

ryanlimdx committed:
Add getResponse method for use with GUI

ryanlimdx committed:
Add greeting on launch

ryanlimdx committed:
Remove print statements when tasks are parsed

ryanlimdx committed:
Merge branch 'branch-GUI'

ryanlimdx committed:
Clean up whitespace

ryanlimdx committed:
Add application to build.gradle

The jar file is on an old version. The old version does not have a GUI.
The application field is pointed to the launcher class that generates a
GUI for user interaction with the app.

Let's push GUI to improve user interaction with the programme.

ZhiWei1010 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,27 @@
+    /**
+     * Exception message template for when creating a task with an empty description.
+     */
+    public static final String NON_EMPTY_DESC = "The description of a %s cannot be empty!";
```
I like that you followed the SCREAMING_SNAKE_CASE here!

ZhiWei1010 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,123 @@
+
+import duke.task.Task;
+/**
+ * The TaskList class represents a list of tasks in the Duke chatbot.
```
I like that you wrote javadocs for all your classes and methods!

ZhiWei1010 commented on src/main/java/duke/utils/Ui.java:
```diff
@@ -0,0 +1,42 @@
+    public static final String LINE =
+            "_________________________________________________________________________\n";
+
+    private static final String[] CATCHPHRASES = {
```
Perhaps the constant variable could be renamed to CATCH_PHRASES following the SCREAMING_SNAKE_CASE?

ZhiWei1010 commented on src/main/java/duke/utils/DateTime.java:
```diff
@@ -0,0 +1,76 @@
+    @Override
+    public String toString() {
+        try {
+            if (datetime != null) {
```
I like how you wrap your if statements in brackets even though it consists of only 1 line!

ZhiWei1010 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,87 @@
+        boolean isExit = Parser.isExit(input);
+
+        // programme
+        while (!isExit) {
```
Is there a reason why you chose to create another variable isExit instead of using Parser.isExit(input) as your conditional check for the while loop?

ZhiWei1010 commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,61 @@
+ */
+public class AddCommand extends Command {
+    private TaskList userTasks;
+    private boolean isAdded;
```
I like how you followed the naming convention for booleans!

ZhiWei1010 commented on src/test/java/duke/StorageTest.java:
```diff
@@ -0,0 +1,35 @@
+
+public class StorageTest {
+    private TaskList taskList;
+    private String path = "./data/testFile.txt";
```
Perhaps path can be defined as a constant?

ZhiWei1010 reviewed (commented):
LGTM. Just a few nits to look through!

ryanlimdx committed:
Add assert statements in Task and Duke

There is no verification that Duke is set up as intended. This may
result in a tasklist not being created, which is the point of the app
(to manage tasks). An assert statement is included in duke to ensure
that a tasklist is created; otherwise, it will prompt the programmer to
create a tasklist. Similarly, assert statements are added to Task to
verify that tasks are marked done (or not done) accurately.

ryanlimdx committed:
Merge branch 'branch-A-Assertions'

ryanlimdx committed:
Add default branch (else) to loadTasks method

ryanlimdx committed:
Merge branch 'branch-A-CodeQuality'

ryanlimdx committed:
Remove global variable

ryanlimdx committed:
Merge branch 'branch-A-CodeQuality'

ryanlimdx committed:
Add ability to check duplicates when adding tasks

ryanlimdx committed:
Merge branch 'branch-BCD-Extension'

ryanlimdx committed:
Abstract task completion check to minimise arrowhead code

ryanlimdx committed:
Abstract task completion checking to minimise arrowhead code in loadTasks method

ryanlimdx committed:
Add assert statements to verify datetime parameters are not null upon initialisation.

ryanlimdx committed:
Add assert statements to verify datetime parameters

ryanlimdx committed:
Merge pull request #2 from ryanlimdx/branch-A-Assertions

Add assert statements to verify datetime parameters are not null upon initialisation

ryanlimdx committed:
Merge branch 'master' into branch-A-CodeQuality

ryanlimdx committed:
Add assert statement to ensure file not null in Storage

ryanlimdx committed:
Merge branch 'master' of https://github.com/ryanlimdx/ip into branch-A-Assertions

ryanlimdx committed:
Merge branch 'branch-A-Assertions'

ryanlimdx committed:
Merge branch 'master' of https://github.com/ryanlimdx/ip into branch-A-CodeQuality

ryanlimdx committed:
Remove print error line for abstraction level

ryanlimdx committed:
Merge branch 'branch-A-CodeQuality'

ryanlimdx committed:
Update header comment for Duke

ryanlimdx committed:
Add README.md

ryanlimdx committed:
Merge branch 'branch-A-UserGuide'

ryanlimdx committed:
Add colons to headers

ryanlimdx committed:
Add storage write when running GUI

ryanlimdx committed:
Update header comment for Duke::main
